MonoGame.Framework/Graphics/Vertices/VertexBuffer.cs

```diff
@@ -26,10 +26,6 @@ protected VertexBuffer(GraphicsDevice graphicsDevice, VertexDeclaration vertexDe
             this.VertexCount = vertexCount;
             this.BufferUsage = bufferUsage;
 
-            // Make sure the graphics device is assigned in the vertex declaration.
-            if (vertexDeclaration.GraphicsDevice != graphicsDevice)
-                vertexDeclaration.GraphicsDevice = graphicsDevice;

```

Comment:
We could live with this maybe...  although I expect XNA does set it.

I wonder how many people have as a hack assumed they could fetch the graphics device via `VertexDeclaration.GraphicsDevice`.